pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -553,14 +557,54 @@ func (os *OpenStack) GetZone() (cloudprovider.Zone, error) {
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (os *OpenStack) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	instanceID, err := instanceIDFromProviderID(providerID)
+	if err != nil {
+		return cloudprovider.Zone{}, err
```

Comment:
can we return `nil, err`?

Reply:
No, nil is incorrect, we cannot use nil as type cloudprovider.Zone in return argument. The "cloudprovider.Zone{}" is ok.

Reply:
The return argument is not a point type, it is a object, nil is incorrect.